src/posterior_gp/approx_posterior_gp.jl

```diff
@@ -107,54 +151,140 @@ function update_approx_posterior(f_post_approx::ApproxPosteriorGP, u::FiniteGP)
-    A_zx = f.data.U' \ cov(f.prior, f.data.z, x)
-    A_zy = f.data.U' \ cov(f.prior, f.data.z, y)
+function Statistics.cov(f::ApproxPosteriorGP{<:VFE}, x::AbstractVector, y::AbstractVector)
+    A_zx = f.data.U' \ cov(f.prior, inducing_points(f), x)
+    A_zy = f.data.U' \ cov(f.prior, inducing_points(f), y)
     return cov(f.prior, x, y) - A_zx'A_zy + Xt_invA_Y(A_zx, f.data.Λ_ε, A_zy)
 end
 
-function StatsBase.mean_and_cov(f::ApproxPosteriorGP{VFE}, x::AbstractVector)
-    A = f.data.U' \ cov(f.prior, f.data.z, x)
+function StatsBase.mean_and_cov(f::ApproxPosteriorGP{<:VFE}, x::AbstractVector)
+    A = f.data.U' \ cov(f.prior, inducing_points(f), x)
     m_post = mean(f.prior, x) + A' * f.data.m_ε
     C_post = cov(f.prior, x) - At_A(A) + Xt_invA_X(f.data.Λ_ε, A)
     return m_post, C_post
 end
 
-function StatsBase.mean_and_var(f::ApproxPosteriorGP{VFE}, x::AbstractVector)
-    A = f.data.U' \ cov(f.prior, f.data.z, x)
+function StatsBase.mean_and_var(f::ApproxPosteriorGP{<:VFE}, x::AbstractVector)
+    A = f.data.U' \ cov(f.prior, inducing_points(f), x)
     m_post = mean(f.prior, x) + A' * f.data.m_ε
     c_post = var(f.prior, x) - diag_At_A(A) + diag_Xt_invA_X(f.data.Λ_ε, A)
     return m_post, c_post
 end
+
+inducing_points(f::ApproxPosteriorGP{<:VFE}) = f.approx.z
+
+"""
+    elbo(v::VFE, fx::FiniteGP, y::AbstractVector{<:Real})
+
+The Titsias Evidence Lower BOund (ELBO) [1]. `y` are observations of `fx`, and `v.z`
+are inducing points.
+
+
+```jldoctest
+julia> f = GP(Matern52Kernel());
+
+julia> x = randn(1000);
+
+julia> z = range(-5.0, 5.0; length=13);
+
+julia> v = VFE(z);
+
+julia> y = rand(f(x, 0.1));
+
+julia> elbo(v, f(x, 0.1), y) < logpdf(f(x, 0.1), y)
+true
+```
+
+[1] - M. K. Titsias. "Variational learning of inducing variables in sparse Gaussian
+processes". In: Proceedings of the Twelfth International Conference on Artificial
+Intelligence and Statistics. 2009.
+"""
+function elbo(v::VFE, fx::FiniteGP, y::AbstractVector{<:Real})
+    _dtc, A = _compute_intermediates(fx, y, fx.f(v.z, v.jitter))
+    return _dtc - (tr_Cf_invΣy(fx, fx.Σy) - sum(abs2, A)) / 2
+end
+
+"""
+    dtc(v::VFE, fx::FiniteGP, y::AbstractVector{<:Real})
+
+The Deterministic Training Conditional (DTC) [1]. `y` are observations of `fx`, and `v.z`
+are inducing points.
+
+
+```jldoctest
+julia> f = GP(Matern52Kernel());
+
+julia> x = randn(1000);
+
+julia> z = range(-5.0, 5.0; length=256);
+
+julia> v = VFE(z);
+
+julia> y = rand(f(x, 0.1));
+
+julia> isapprox(dtc(v, f(x, 0.1), y), logpdf(f(x, 0.1), y); atol=1e-6, rtol=1e-6)
+true
+```
+
+[1] - M. Seeger, C. K. I. Williams and N. D. Lawrence. "Fast Forward Selection to Speed Up
+Sparse Gaussian Process Regression". In: Proceedings of the Ninth International Workshop on
+Artificial Intelligence and Statistics. 2003
+"""
+function dtc(v::VFE, fx::FiniteGP, y::AbstractVector{<:Real})
+    return first(_compute_intermediates(fx, y, fx.f(v.z, v.jitter)))
```

Comment:
How about the more explicit
```suggestion
    _dtc, _ = _compute_intermediates(fx, y, fx.f(v.z, v.jitter))
    return _dtc
```
?